phoenix-core-server/src/main/java/org/apache/phoenix/coprocessor/GlobalIndexRegionScanner.java

```diff
@@ -273,11 +278,31 @@ public GlobalIndexRegionScanner(final RegionScanner innerScanner,
             verificationResultRepository =
                     new IndexVerificationResultRepository(indexMaintainer.getIndexTableName(), hTableFactory);
             nextStartKey = null;
-            minTimestamp = scan.getTimeRange().getMin();
         }
+        computeMinTimestamp(config);
     }
 
-
+    /**
+     * For CDC indexes we do not need to consider rows outside max lookback window or before
+     * the index create time. minTimestamp needs to be computed and used for CDC indexes always
+     * even when it is not set on the scan
+     */
+    private void computeMinTimestamp(Configuration config) throws IOException {
+        minTimestamp = scan.getTimeRange().getMin();
+        if (indexMaintainer.isCDCIndex()) {
+            minTimestamp = EnvironmentEdgeManager.currentTimeMillis() - maxLookBackInMills;
```

Comment:
Currently, `maxLookback` is used only during Index verification and thus, table level max lookback is passed as scan attribute only when scan attributes: `INDEX_REBUILD_VERIFY_TYPE` and `REBUILD_INDEXES` are also set. But by shifting dependency on max lookback outside index verification we now need to pass table level max lookback as scan attribute wherever we are setting `REBUILD_INDEXES`. Example: In `GlobalIndexChecker.GlobalIndexScanner.repairIndexRows()`.

Reply:
ServerBuildIndexCompiler is used for rebuild and verification and we set the scan attribute there. GlobalIndexChecker does one row repair for covered indexes. We do not need to set there as it is not used for uncovered indexes and thus CDC indexes.

Reply:
Got it, thanks. I saw that `repairIndexRows()` is called for covered indexes only but missed the part that CDC indexes are uncovered. Thanks

Reply:
`MetadataRegionObserver.BuildIndexSchedulerTask.run()` seems like the only other class which can be on Index rebuild path. This also sets `REBUILD_INDEXES` scan attribute but is not setting table level max lookback as scan attribute. Please correct me if I am wrong. Thanks

Reply:
That is for old indexes. The new indexes does use that code path.

Reply:
@kadirozde I guess you mean that `MetadataRegionObserver.BuildIndexSchedulerTask.run()` isn't used anymore with new indexes? (Above sentence seems a bit confusing so, just confirming for my understanding).

Reply:
The old index design on an index write failure, transitions indexes to the PENDING_DISABLE state on the server side and after the client retries, and after retries are exhausted, disables indexes. MetadataRegionObserver rebuilds these indexes incrementally. The new design does not change the state of indexes on index write failure as these failures are handled by read repair. After upgrading to new indexes, phoenix.index.failure.handling.rebuild can be set to false to skip running BuildIndexScheduleTask. This is what we do in production.